### PR TITLE
Missing applicationUrl does not inherit dashboard url

### LINKED ISSUE #225: Service application gets dashboard url injected if there's no launch settings
Make an ASP.NET Core application and delete the launch settings. Run the app host and observe that the application fails to run. This is because applicationUrl from the app host is being injected as ASPNETCORE_URLS which flows to the child process.

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/dotnet/aspire/issues/225

### CODE REVIEW

File: src/Aspire.Hosting/Dcp/ApplicationExecutor.cs [hunk 1]
```diff
@@ -401,19 +401,22 @@ private async Task CreateExecutablesAsync(IEnumerable<AppResource> executableRes
...
                 // Need to apply configuration settings manually; see PrepareExecutables() for details.
                 if (er.ModelResource is ProjectResource project && project.SelectLaunchProfileName() is { } launchProfileName && project.GetLaunchSettings() is { } launchSettings)
                 {
                     ApplyLaunchProfile(er, config, launchProfileName, launchSettings);
+                    hasLaunchProfile = true;
                 }
-                else
+
+                // We want to make sure that certain environment variables are NOT inherited
+                foreach (var envVar in s_doNotInheritEnvironmentVars)
```
Comment (BrennanConroy) [inline src/Aspire.Hosting/Dcp/ApplicationExecutor.cs:413]: Actually, should this just move to after `if (er.ServicesProduced.Count > 0) {}`? That way if no one sets the forbidden env variables we'll set empty ones.

**Comment by davidfowl** [comment]
How do you know it’s working without a test 👀

**Comment by BrennanConroy** [comment]
I tried to write a test, but it's kinda difficult... how do you get the endpoint info for the app? Users can find it by looking at the logs.

**Review by davidfowl — APPROVED** [review]

**Comment by davidfowl** [comment]
I think this test would be exposing an end point that dumps config or all env variables. The test would look at the effective config.

**Pull request merged**